storagenode/notifications/notifications.go

```diff
@@ -0,0 +1,70 @@
+package notifications
+
+import (
+	"context"
+	"time"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
+
+	"storj.io/storj/pkg/storj"
+)
+
+// DB is an implementation of notifications.Notifications.
+//
+// architecture: Database
+type DB interface {
+	Insert(ctx context.Context, notification NewNotification) (Notification, error)
+	List(ctx context.Context, cursor NotificationCursor) (NotificationPage, error)
+	Read(ctx context.Context, notificationID uuid.UUID) error
+	ReadAll(ctx context.Context) error
+}
+
+// NotificationType is a numeric value of specific notification type.
+type NotificationType int
+
+const (
+	// NotificationTypeCustom is a common notification type which doesn't describe node's core functionality.
+	NotificationTypeCustom NotificationType = 0
+	// NotificationTypeAuditCheckFailure is a notification type which describes node's audit check failure.
+	NotificationTypeAuditCheckFailure NotificationType = 1
+	// NotificationTypeUptimeCheckFailure is a notification type which describes node's uptime check failure.
+	NotificationTypeUptimeCheckFailure NotificationType = 2
+	// NotificationTypeDisqualification is a notification type which describes node's disqualification status.
+	NotificationTypeDisqualification NotificationType = 3
+)
+
+// NewNotification holds notification entity info which is being inserted to database.
+type NewNotification struct {
+	SenderID storj.NodeID
+	Type     NotificationType
+	Title    string
+	Message  string
+}
+
+// Notification holds notification entity info which is being retrieved from database.
+type Notification struct {
+	ID        uuid.UUID
```

Comment:
should we inherit Notification from the NewNotification?

Reply:
let's leave it as it is for better readability

Reply:
let's leave it as it is for better readability